processor/parse/utils.js

```diff
@@ -0,0 +1,21 @@
+function insertTokenizerBefore({ name, before, tokenizer, type = 'block' }) {
+  const { Parser } = this;
+  const tokenizers = Parser.prototype[`${type}Tokenizers`];
+  const methods = Parser.prototype[`${type}Methods`];
+
+  const index = methods.indexOf(before);
+  if (index === -1) {
+    throw new Error(`The '${before}' tokenizer doesn't exist!`);
+  }
+
+  tokenizers[name] = tokenizer;
+  methods.splice(index, 0, name);
+}
+
+export function insertBlockTokenizerBefore(args) {
+  insertTokenizerBefore.call(this, args);
+}
+
+export function insertInlineTokenizerBefore(args) {
+  return insertTokenizerBefore.call(this, { ...args, type: 'inline' });
+}
```

Comment:
I think we should add a test for this if we're centralizing all the `.splice` logic in to a single place

Reply:
Hmm, like test this specific function?